includes/attendee/attendee-repository.php

```diff
@@ -99,73 +106,76 @@ public function get_attendee( int $event_id, int $user_id ): ?Attendee {
 			return null;
 		}
 
-		$attendee = new Attendee( $event_id, $row->user_id );
-		if ( '1' === $row->is_host ) {
-			$attendee->mark_as_host();
-		}
-
-		return $attendee;
-	}
-
-	/**
-	 * @return Attendee[] Attendees of the event.
-	 */
-	public function get_attendees( int $event_id ): array { // phpcs:ignore Generic.CodeAnalysis.UnusedFunctionParameter.Found
-		// TODO.
-		return array();
+		return new Attendee(
+			$event_id,
+			$row->user_id,
+			'1' === $row->is_host,
+			null === $row->locales ? array() : explode( ',', $row->locales ),
+		);
 	}
 
 	/**
-	 * Get attendees without contributions for an event.
+	 * Retrieve all the attendees of an event.
 	 *
-	 * @return Attendee[]
+	 * @return Attendee[] Attendees of the event.
 	 * @throws Exception
 	 */
-	public function get_attendees_not_contributing( int $event_id ): array {
+	public function get_attendees( int $event_id ): array {
 		global $wpdb, $gp_table_prefix;
 
 		// phpcs:disable WordPress.DB.PreparedSQL.InterpolatedNotPrepared
 		// phpcs:disable WordPress.DB.DirectDatabaseQuery.DirectQuery
 		// phpcs:disable WordPress.DB.DirectDatabaseQuery.NoCaching
-		$all_attendees = $wpdb->get_results(
+		$rows = $wpdb->get_results(
 			$wpdb->prepare(
 				"
-				select user_id, is_host
-				from {$gp_table_prefix}event_attendees
+				select
+					user_id,
+					is_host,
+					(
+						select group_concat( distinct locale )
+						from {$gp_table_prefix}event_actions
+						where event_id = attendees.event_id
+						  and user_id = attendees.user_id
+					) as locales
+				from {$gp_table_prefix}event_attendees attendees
 				where event_id = %d
 			",
 				array(
 					$event_id,
 				)
-			),
+			)
 		);
+		// phpcs:enable
 
-		$contributing_ids = $wpdb->get_col(
-			$wpdb->prepare(
-				"
-				select distinct user_id
-				from {$gp_table_prefix}event_actions
-				where event_id = %d
-			",
-				array(
+		return array_map(
+			function ( $row ) use ( $event_id ) {
+				return new Attendee(
 					$event_id,
-				)
-			)
+					$row->user_id,
+					'1' === $row->is_host,
+					null === $row->locales ? array() : explode( ',', $row->locales ),
+				);
+			},
+			$rows,
 		);
-		// phpcs:enable
+	}
 
-		$attendees_not_contributing = array();
-		foreach ( $all_attendees as $attendee_row ) {
-			if ( ! in_array( $attendee_row->user_id, $contributing_ids, true ) ) {
-				$attendee = new Attendee( $event_id, $attendee_row->user_id );
-				if ( '1' === $attendee_row->is_host ) {
-					$attendee->mark_as_host();
+	/**
+	 * Get attendees without contributions for an event.
+	 *
+	 * @return Attendee[]
+	 * @throws Exception
+	 */
+	public function get_attendees_not_contributing( int $event_id ): array {
+		return array_values(
```

Comment:
If we opt to use user_ids as array keys (see above), we could forgo this.
```suggestion
		return 
```